eng/common/TestResources/Remove-TestResources.ps1

```diff
@@ -47,7 +47,11 @@ param (
     [string] $Environment = 'AzureCloud',
 
     [Parameter()]
-    [switch] $Force
+    [switch] $Force,
+
+    # Captures any arguments not declared here (no parameter errors)
+    [Parameter(ValueFromRemainingArguments = $true)]
+    $RemoveTestResourcesRemainingArguments
```

Comment:
If other scripts are using `ValueFromRemainingArguments` and it has the same name (e.g. `$RemainingArguments` in both this script and the script being called) then the value in this script will simply be passed in and the remaining variables will cause the script being called to throw an error about extra parameters. 

There are two obvious solutions here: 
1) Name the parameters uniquely in each script to reduce the probability of this kind of collision 
1) Remove the remaining arguments variable from the parameters that are splat into the called script

Reply:
I'd just call this `AdditionalArguments` in case we ever want to use it later. We don't have to use it now, but it's consistently named.

Reply:
I'm not entirely sure I understand why this is needed?

Reply:
Because you pass the same splatted args into the removal script, which doesn't have all the same parameters. Splatting errs on more parameters than declared. This effectively "eats" them.

Reply:
OK I figured Powershell would just add them to the unbound set of parameters.

Reply:
No such thing unless you explicitly opt into it this way for `CmdletBinding` (advanced functions), which provide a lot of benefits over regular functions.

Reply:
The reason I'm doing this here is because we already have at least one example of using `RemainingArguments` in a `pre-` script... This prevents an interface break.